src/Backend/Core/Js/Components/Forms.js

```diff
@@ -0,0 +1,403 @@
+
+        // only do something when the current value and the placeholder are the same
+        if ($input.val() === $input.attr('placeholder')) {
+          // clear
+          $input.val('')
+
+          // remove class
+          $input.removeClass('placeholder')
+        }
+      })
+
+      $placeholder.blur((event) => {
+        // grab element
+        const $input = $(event.currentTarget)
+
+        // only do something when the input is empty or the value is the same as the placeholder
+        if ($input.val() === '' || $input.val() === $input.attr('placeholder')) {
+          // set placeholder
+          $input.val($input.attr('placeholder'))
+
+          // add class
+          $input.addClass('placeholder')
+        }
+      })
+
+      // call blur to initialize
+      $placeholder.blur()
+
+      // hijack the form so placeholders aren't submitted as values
+      $placeholder.parents('form').submit((event) => {
+        // find elements with placeholders
+        $(event.currentTarget).find('input[placeholder]').each((index, element) => {
+          // grab element
+          const $input = $(element)
+
+          // if the value and the placeholder are the same reset the value
+          if ($input.val() === $input.attr('placeholder')) $input.val('')
+        })
+      })
+    }
+  }
+
+  // replaces buttons with <a><span>'s (to allow more flexible styling) and handle the form submission for them
+  submitWithLinks () {
+    // the html for the button that will replace the input[submit]
+    const replaceHTML = '<a class="{class}" href="#{id}"><span>{label}</span></a>'
+
+    // are there any forms that should be submitted with a link?
+    if ($('form.submitWithLink').length > 0) {
+      $('form.submitWithLink').each((index, element) => {
+        // get id
+        const formId = $(element).attr('id')
+        let dontSubmit = false
+
+        // validate id
+        if (formId !== '') {
+          // loop every button to be replaced
+          $('form#' + formId + '.submitWithLink input[type=submit]').each((index, element) => {
+            $(element).after(replaceHTML.replace('{label}', $(element).val()).replace('{id}', $(element).attr('id')).replace('{class}', 'submitButton button ' + $(element).attr('class'))).css({
+              position: 'absolute',
+              top: '-9000px',
+              left: '-9000px'
+            }).attr('tabindex', -1)
+          })
+
+          // add onclick event for button (button can't have the name submit)
+          $('form#' + formId + ' a.submitButton').on('click', (event) => {
+            event.preventDefault()
+
+            // is the button disabled?
+            if ($(event.currentTarget).prop('disabled')) {
+              return false
+            }
+            else {
+              $('form#' + formId).submit()
+            }
+          })
+
+          // dont submit the form on certain elements
+          $('form#' + formId + ' .dontSubmit').on('focus', () => {
```

Comment:
With ES6 template literals we can do this now? I personally find it more readable, not sure what you think?

```js
$(`form#${formId} .dontSubmit`).on('focus', () => {
```